### PR TITLE
Moving xen-tools into a separate service

### PR DESCRIPTION
**Pull request by rvs**
This is a final PR in a series that I've been working on. This one is probably the most interesting one and the one that can affect EVE (hence extra reviews and testing are required). There's may be quite a bit going on in this PR so I suggest reviewing it, perhaps, commit by commit.

So here are the highlights of these commits and why they are required:
   * It took me some time to make sure xen-tools and pillar can still communicate effectively. For now, I'm simply relying on domainmanager calling "fake" xl, xenstore and xentop commands that simply use ctr to exec real commands inside of the xen-tools container https://github.com/lf-edge/eve/pull/625/commits/36ab8e2656fa8fe27ab30b48994531ed77700090
   * But that wasn't all that was required: since rkt insists on putting together a rootfs for containers it executes via bind mounts inside of the pillar container I had to make sure those bind mounts would be globally visible AND that root filesystem permissions in the pillar container would correspond to the ones in the xen-tools container https://github.com/lf-edge/eve/pull/625/commits/e3922280c1d325f97d6123a9c4780b69d22cf8cf
   * A few other filesystem-based workaround for rkt were also required and hence this commit was made: https://github.com/lf-edge/eve/pull/625/commits/d324a8624824ae764ea61eecdbc27a25367c3cfc
    * Finally, I bit the bullet and made Xen log directly to memlogd https://github.com/lf-edge/eve/pull/625/commits/68cf3a4cc95e661ea5fed562874930a3be4cfee1 -- this basically required creating a UNIX FIFO (instead of opening a file in a filesystem) and sending one end of it to memlogd via UNIX domain socket
    * And took advantage of it by removing all file-based logging from pillar and xen-tools (greatly simplifying our logging infrastructure in the process) https://github.com/lf-edge/eve/pull/625/commits/44be7e16edf9daae8c06b59b8832f7a4257b3d2c

The last two are kind of interesting, because we now have full logging from all of our domains (VMs and Containers) streamed into memlogd. So it is only a matter of time when @gkodali-zededa picks them up in rsyslog.

Btw, for now -- we're using a single memlogd for both system-level logs and domain-level logs, but if we like this architecture we can run a 2nd one just for domains (after all -- domains do get a separate logging treatment from the cloud side @zedvijay)

### CODE REVIEW

**Comment by rvs** [comment]
Yo! Nice ppl @kalyan-nidumolu @nordmark @deitch -- you kind of have to review this ;-)

**Comment by deitch** [comment]
There appears to be a lot of "cross-plumbing" here: 

* `xen-tools` mounting `/:/hostfs`
* domainmgr in pillar calling `/usr/sbin/xl` (that part is normal) in pillar, which in turn is calling `exec chroot /hostfs ctr --namespace ...` (that one is a doozy)
* `/persist/rkt-cni`

I might have missed a few

Do we lose some of the benefits of splitting it out? This is hard enough to understand (not everyone is @rvs!). 

I have a suggestion. Can we write a "flow doc", something that describes, "here is container pillar, here is container xen-tools, here is what the first does, here is what the second does, and here are the flows between them and how they happen." That might make it much easier to understand (and review), and possibly make improvements going forward.

> rkt insists on putting together a rootfs for containers it executes via bind mounts inside of the pillar container

I didn't understand that one. rkt unpacks images, then creates a rootfs for the container. Where do the bind-mounts come in, and how do they affect pillar vs xen-tools containers?

File: pkg/pillar/rootfs/usr/sbin/xl [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+# shellcheck disable=SC2046,SC2086
+exec chroot /hostfs /usr/bin/ctr --namespace services.linuxkit t exec --cwd / ${TERM:+-t} --exec-id $(basename $(mktemp)) xen-tools env ${TERM:+TERM=}$TERM $(basename $0) "$@"
```
Comment (deitch) [inline pkg/pillar/rootfs/usr/sbin/xl]: This seems convoluted. Any reason not to just bind-mount the `containerd.sock` file into pillar, and then do:

```
ctr --address /run/containerd/containerd.sock --namespace services.linuxkit t exec --cwd / ${TERM:+-t} --exec-id $(basename $(mktemp)) xen-tools env ${TERM:+TERM=}$TERM $(basename $0) "$@"
```

Actually, isn't `/run` already mounted in in `build.yml`? Why do we even need to `chroot` to do it?

**Comment by rvs** [comment]
> With this xl command calling into the xen-tools container do you still want the pubsub way to invoke commands?

More than ever @nordmark -- that's actually my answer to @deitch Speaking of which:

> There appears to be a lot of "cross-plumbing" here:

@deitch -- some of that plumbing is a stop-gap waiting to go away. Perhaps a fair request would be to document it in some file/README in xen-tools. More specifically:
    * once we get rid of rkt
    * once @nordmark helps me with a exec-over-pubsub service
A lot of this plumbing will go away

**Comment by rvs** [comment]
Hey @nordmark I believe that on Linux data exchange under PIPE_BUF size is in fact atomic. PIPE_BUF is 4k which should be more than enough for this use case. If you have conflicting evidence, can you please point that out?

**Pull request merged**